### PR TITLE
Add a new "Dependencies and Internal Tools" section in the docs

### LINKED ISSUE #310: Add ScanCode.io Dependencies section to Reference Documentation
Converted from @mjherzog card:

Add ScanCode/AboutCode Dependencies section to Reference Documentation - we need to explain more clearly the "internal" dependencies of ScanCode.io on ScanCode Toolkit, container-inspector and other nexB-sponsored projects. One purpose is to help users understand the release pipeline for a new version of SCIO.

### PR DESCRIPTION
**Pull request by tdruez**
Closes #310

Signed-off-by: Thomas Druez <tdruez@nexb.com>

### CODE REVIEW

**Comment by tdruez** [comment]
Hi @hyounes4560 could you review and edit the new chapter content?

File: docs/introduction.rst [hunk 1]
```diff
@@ -83,6 +75,38 @@ container images and virtual machine (VM) disk images that often harbor
...
+- :ref:`rest_api`
+- :ref:`command_line_interface`
+
+.. note::
+    Multiple applications from the Django eco-system are also included,
+    see the `setup.cfg <https://github.com/nexB/scancode.io/blob/main/setup.cfg>`_ file
+    for an exhaustive list of dependencies.
+
+The second essential part of ScanCode.io is the **ScanCode Toolkit**, used for the
+extraction of archives and the scanning engine.
```
Comment (hyounes4560) [inline docs/introduction.rst:98]: The second essential part of ScanCode.io is the **ScanCode Toolkit**, which is used for archives extraction and as the scanning engine.

File: docs/introduction.rst [hunk 2]
```diff
@@ -83,6 +75,38 @@ container images and virtual machine (VM) disk images that often harbor
...
+
+The nexB `container-inspector <https://github.com/nexB/container-inspector>`_ library
+is also a key component of ScanCode.io as the tool used for the analysis of Docker
+images, containers, root filesystems and virtual machine images.
+
```
Comment (hyounes4560) [inline docs/introduction.rst:103]: The nexB `container-inspector <https://github.com/nexB/container-inspector>`_ library
is also a key component of ScanCode.io as this tool is used to analyse Docker
images, containers, root filesystems, and virtual machine images.

File: docs/introduction.rst [hunk 3]
```diff
@@ -83,6 +75,38 @@ container images and virtual machine (VM) disk images that often harbor
...
+
+.. note::
+    Releases of ScanCode.io usually follow the releases of ScanCode Toolkit to ensure
+    the latest improvements of the scanning engines are available as soon as ready.
+
```
Comment (hyounes4560) [inline docs/introduction.rst:108]: As a common practice, ScanCode.io releases usually follow ScanCode Toolkit releases to ensure
    the latest improvements of the scanning engines are included in the latest release of ScanCode.io.

**Review by hyounes4560 — CHANGES_REQUESTED** [review]
FYI @tdruez

**Pull request merged**